src/GraphQL/Validation/BasicVisitor.cs

```diff
@@ -30,36 +29,33 @@ public BasicVisitor(IList<INodeVisitor> visitors)
         }
 
         /// <summary>
-        /// Walks the specified <see cref="INode"/>, executing <see cref="INodeVisitor.Enter(INode, ValidationContext)"/> and
-        /// <see cref="INodeVisitor.Leave(INode, ValidationContext)"/> methods for each node.
+        /// Walks the specified <see cref="ASTNode"/>, executing <see cref="INodeVisitor.Enter(ASTNode, ValidationContext)"/> and
+        /// <see cref="INodeVisitor.Leave(ASTNode, ValidationContext)"/> methods for each node.
         /// </summary>
-        public void Visit(INode node, ValidationContext context) => VisitRecursive(node, new State(context, _visitors));
-
-        private static void VisitRecursive(INode node, State state)
+        public override async ValueTask Visit(ASTNode? node, State context)
```

Comment:
Method is now async, and should be named `VisitAsync`. Can fix in this PR or separate, but it should be fixed.

Reply:
It overrides base method. All methods from `DefaultNodeVisitor`/`INodeVisitor` are async. I do not want to rename them all (58). I saw a recommendation from MS somewhere that methods do not necessarily should have async suffix if initially the entire class was designed asynchronous.

Reply:
No problem; I will do the renaming.

https://github.com/graphql-dotnet/parser/pull/209